docs/spec/types.md

```diff
@@ -0,0 +1,78 @@
+
+# Bicep Type System
+Because programs written in Bicep transpile primarily into JSON, it was clear from the beginning that Bicep has to support all data types that are available in JSON. We also wanted to provide stronger typing and validation so the users' errors are caught early in the development process and more failed deployments are avoided. These goals are very similar to the goals of the [TypeScript](https://www.typescriptlang.org/) language, which adds strong typing to JavaScript and compiles into the same.
+
+As a result, the Bicep type system implements a small subset of the capabilities present in the TypeScript type system that are needed to support Bicep scenarios. If you are familiar with TypeScript and JSON, many of these concepts will feel familiar.
+
+## Data Types
+Bicep has a [Structural type system](https://en.wikipedia.org/wiki/Structural_type_system). In other words, the assignability of types in Bicep is determined by the definition (or structure) of the type rather than its name. All types in Bicep do have names, but these are used as shorthand type identifiers in error messages and hover texts.
+
+The Bicep type system automatically infers the types of expressions but does not perform any automatic type conversion or coercion. Any such operations must be declared explicitly.
+
+All values in Bicep have one of the following types:
+
+### Simple Types
+| Name | Description |
+|:-|:-|
+| `any` | The value can be anything or allows anything. |
+| `error` | The expression has an error. |
+| `string` | Represents arbitrary text. Equivalent to JSON strings. |
+| `number` | **Not yet implemented (#486)** Represents a floating point number or a big integer. Equivalent to the JSON 
```

Comment:
GitHub doesn't make the `#` a link in docs like this. Can you add the link manulaly?

Reply:
Didn't realize that. Will fix...

Reply:
Fixed